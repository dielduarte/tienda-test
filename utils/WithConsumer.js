import React, { createFactory, Component } from 'react';
import { setDisplayName, wrapDisplayName } from 'recompose';
import { StoreContext } from '../store';

const WithConsumer = BaseComponent => {
  const factory = createFactory(BaseComponent);

  class WithConsumer extends Component {
    render() {
      return (
        <StoreContext.Consumer>
          {({ store, dispatch }) =>
            factory({
              ...{ ...this.props, store, dispatch },
              ...this.state
            })
          }
        </StoreContext.Consumer>
      );
    }
  }

  if (process.env.NODE_ENV !== 'production') {
    return setDisplayName(wrapDisplayName(BaseComponent, 'WithConsumer'))(
      WithConsumer
    );
  }
  return WithConsumer;
};

export default WithConsumer;
