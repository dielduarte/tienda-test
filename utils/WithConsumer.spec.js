import React from 'react';
import WithConsumer from './WithConsumer';
import { shallow } from 'enzyme';

function FakeComponent() {
  return <h1>Text</h1>;
}

describe('WithConsumer', () => {
  it('sould return the wrappedComponent with access to store object and dispatch method', () => {
    const Component = WithConsumer(FakeComponent);
    const wrapper = shallow(<Component />);

    expect(wrapper).toMatchSnapshot();
  });
});
