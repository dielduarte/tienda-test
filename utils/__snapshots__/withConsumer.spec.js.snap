// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithConsumer sould return the wrappedComponent with access to store object and dispatch method 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
